solidity/contracts/bridge/Redeem.sol

```diff
@@ -0,0 +1,848 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.9;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+import {BytesLib} from "@keep-network/bitcoin-spv-sol/contracts/BytesLib.sol";
+
+import "./BitcoinTx.sol";
+import "./BridgeState.sol";
+import "./Wallets.sol";
+
+import "../bank/Bank.sol";
+
+/// @notice Aggregates functions common to the redemption transaction proof
+///         validation and to the moving funds transaction proof validation.
+library OutboundTx {
```

Comment:
Do we plan to leave that library here?

Reply:
I think so. I put it into a separate file but it does not look good next to `BitcoinTx`. It's not really a separate library as I see it. It's a part of the redemption code that is reused by moving funds logic (moving funds is a kind of redemption) - that's how I see it.

Reply:
> It's a part of the redemption code that is reused by moving funds logic (moving funds is a kind of redemption) - that's how I see it.

Yep, that's my reasoning too. Anyway, that's non-blocking for this PR. We can decide about that library at the end.